lib/ansible/plugins/action/shell.py

```diff
@@ -17,8 +17,10 @@ def run(self, tmp=None, task_vars=None):
 
         # Shell shares the same module code as command. Fail if command
         # specific options are set.
-        if "expand_argument_vars" in self._task.args:
-            raise AnsibleActionFail(f"Unsupported parameters for ({self._task.action}) module: expand_argument_vars")
+        invalid_args = ("expand_argument_vars", "warn")
+        for invalid_arg in invalid_args:
+            if invalid_arg in self._task.args:
+                raise AnsibleActionFail(f"Unsupported parameters for ({self._task.action}) module: {invalid_arg}")
```

Comment:
When there's several invalid args, this would only show one problem. May I suggest listing all of the violations?

```suggestion
        invalid_arg_candidates = {"expand_argument_vars", "warn"}
        if invalid_args := invalid_arg_candidates & set(self._task.args):
            raise AnsibleActionFail(
                f"Unsupported parameters for ({self._task.action}) "
                f"module: {', '.join(invalid_args)}.",
            )
```

Reply:
the issue is not 'warn' itself, but that the error is issued by 'command' and not 'shell'. Instead of fixing one instance, fix all of them: have command use 'caller name' if it exists instead of own.